File: src/test/java/org/appsugar/archetypes/controller/UserControllerTest.java
package org.appsugar.archetypes.controller;

import com.cosium.spring.data.jpa.entity.graph.domain.EntityGraph;
import com.querydsl.core.types.OrderSpecifier;
import lombok.SneakyThrows;
import lombok.extern.slf4j.Slf4j;
import org.appsugar.archetypes.domain.User;
import org.appsugar.archetypes.repository.UserRepository;
import org.junit.Ignore;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;
import org.springframework.boot.test.mock.mockito.MockBean;
import org.springframework.security.test.context.support.WithMockUser;
import org.springframework.test.web.servlet.MvcResult;

import java.util.Arrays;
import java.util.List;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

@Disabled
@Slf4j
public class UserControllerTest extends BaseControllerTest {
    @MockBean
    private UserRepository userRepository;


    @Test
    @SneakyThrows
    @WithMockUser(authorities = "user:list")
    public void testList() {
        List<User> expectedUsers = Arrays.asList(new User(1l, "admin", "admin", "admin", "123456", null));

        Mockito.when(userRepository.findAll(Mockito.any(EntityGraph.class), Mockito.any(OrderSpecifier[].class))).thenReturn(expectedUsers);

        MvcResult mvcResult = mockMvc.perform(get(UserController.LIST_URL).contentType(CONTENT_TYPE_JSON)).andExpect(status().isOk()).andReturn();
        String result = mvcResult.getResponse().getContentAsString();
        String expectedResult = objectMapper.writeValueAsString(expectedUsers);
        log.debug("testList expectedUsers is {}  real users is {}", expectedResult, result);
        // Assertions.assertEquals(expectedResult, result);
    }
}

File: src/test/kotlin/org/appsugar/archetypes/controller/UserControllerTest.kt
package org.appsugar.archetypes.controller

import kotlinx.coroutines.runBlocking
import org.appsugar.archetypes.BaseControllerTestCase
import org.appsugar.archetypes.TypedPage
import org.appsugar.archetypes.TypedResponse
import org.appsugar.archetypes.await
import org.appsugar.archetypes.common.domain.Response
import org.appsugar.archetypes.entity.User
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

class UserControllerTest : BaseControllerTestCase() {
    lateinit var userFacade: UserFacade

    @Test
    fun testList() = runBlocking {
        val response = userFacade.list().await()!!
        logger.debug("testList result is {}", response)
        Assertions.assertEquals(Response.SUCCESS.code, response.code)
    }

    @Test
    fun testForm() = runBlocking {
        val response = userFacade.form(-1).await()!!
        logger.debug("test form result is {}", response.data)
    }

    override fun postConstruct() {
        super.postConstruct()
        userFacade = buildFacade(UserFacade::class.java)
    }
}

interface UserFacade {
    @GET("/system/user/list")
    fun list(): Call<TypedResponse<TypedPage<User>>>

    @GET("/system/user/detail")
    fun form(@Query("id") id: Long): Call<TypedResponse<User>>
}